src/psyclone/transformations.py

```diff
@@ -3306,21 +3307,25 @@ def validate(self, kern, options=None):
             raise TransformationError("Cannot transform kernel {0} because "
                                       "it will be module-inlined.".
                                       format(kern.name))
+        if kern.modified:
+            raise TransformationError(
+                "Cannot transform kernel '{0}' because it has previously been "
+                "transformed and this transformation works on the fparser2 "
+                "parse tree rather than the PSyIR (#490).".format(kern.name))
 
         # Perform general validation checks. In particular this checks that
-        # a PSyIR of the kernel body can be constructed.
+        # the PSyIR of the kernel body can be constructed.
         super(ACCRoutineTrans, self).validate(kern, options)
 
-        # Check that the kernel does not access any data via a module 'use'
-        # statement
+        # Check that the kernel does not access any data or routines via a
+        # module 'use' statement
         sched = kern.get_kernel_schedule()
         global_variables = sched.symbol_table.global_datasymbols
         if global_variables:
             raise TransformationError(
                 "The Symbol Table for kernel '{0}' contains the following "
-                "symbols with 'global' scope: {1}. PSyclone cannot currently "
-                "transform kernels for execution on an OpenACC device if "
-                "they access data not passed by argument.".
+                "symbol(s) with global scope: {1}. PSyclone cannot currently "
```

Comment:
I've updated the error message. As it happens, the two cases are different since I think we simply can't call another kernel from within an OpenCL kernel whereas we have a ticket to support this for OpenACC.